xla/stream_executor/rocm/BUILD

```diff
@@ -349,7 +351,7 @@ cc_library(
         "@com_google_absl//absl/strings",
         "@com_google_absl//absl/types:span",
         "@local_config_rocm//rocm:rocm_headers",
-        "@tsl//tsl/platform:env",
+        "@tsl//tsl/platform:env_impl",
```

Comment:
@ddunl do you know why it might be needed here? I suspect "something" might be missing in one of the build files to add it automatically?

Reply:
This strikes me as odd - I'd expect this to be taken care of by https://github.com/openxla/xla/blob/main/xla/xla.bzl#L52 for all test/binary targets. @hsharsha if you have a build failure log for what happens when you switch this back to `:env` I'd be very interested to see it

Reply:
@ddunl Running local_client_aot_test after switching back to `:env` with command line 
```
bazel  --bazelrc=/usertools/rocm.bazelrc test  --test_output all --config=rocm --test_env=TF_TESTS_PER_GPU=1  --run_under=//tools/ci_build/gpu_build:parallel_gpu_execute -- //xla/tests:local_client_aot_test
```
Results in this error
```
INFO: Analyzed target //xla/tests:local_client_aot_test (158 packages loaded, 12668 targets configured).
INFO: Found 1 test target...
ERROR: /home/hahavanu/xla_upstream/upstream/xla/xla/tests/BUILD:2441:12: Linking xla/tests/local_client_aot_test failed: (Exit 1): crosstool_wrapper_driver_is_not_gcc failed: error executing command (from target //xla/tests:local_client_aot_test) external/local_config_rocm/crosstool/clang/bin/crosstool_wrapper_driver_is_not_gcc @bazel-out/k8-opt/bin/xla/tests/local_client_aot_test-2.params
bazel-out/k8-opt/bin/external/tsl/tsl/profiler/backends/cpu/_objs/traceme_recorder_impl/traceme_recorder.o:traceme_recorder.cc:function tsl::profiler::TraceMeRecorder::Record(tsl::profiler::TraceMeRecorder::Event&&): error: undefined reference to 'tsl::Env::Default()'
bazel-out/k8-opt/bin/xla/stream_executor/rocm/_objs/miopen_plugin/rocm_dnn.o:rocm_dnn.cc:function stream_executor::gpu::wrap::DynLoadShim__miopenDestroy::LoadOrDie(): error: undefined reference to 'tsl::Env::Default()'
bazel-out/k8-opt/bin/xla/stream_executor/rocm/_objs/miopen_plugin/rocm_dnn.o:rocm_dnn.cc:function stream_executor::gpu::MIOpenAccess::GetHandle(stream_executor::gpu::GpuExecutor*, stream_executor::Stream*): error: undefined reference to 'tsl::Env::Default()'
bazel-out/k8-opt/bin/xla/stream_executor/rocm/_objs/miopen_plugin/rocm_dnn.o:rocm_dnn.cc:function stream_executor::gpu::MIOpenCTCLossDescriptor::MIOpenCTCLossDescriptor(miopenDataType_t): error: undefined reference to 'tsl::Env::Default()'
collect2: error: ld returned 1 exit status
Target //xla/tests:local_client_aot_test failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 582.787s, Critical Path: 232.26s
INFO: 4791 processes: 263 internal, 4528 local.
FAILED: Build did NOT complete successfully
//xla/tests:local_client_aot_test                               FAILED TO BUILD
```